### ISSUE TITLE
LogEntryEvent__e.HttpRequestEndpoint__c too short to store endpoint

### ISSUE THREAD

**Issue by ILambuRI** [jongpie/NebulaLogger#768]
### Package Edition of Nebula Logger

Unlocked Package

### Package Version of Nebula Logger

v4.13.10

### New Bug Summary

I send a HTTP request with long get parameters in the URL and I am receiving this email:

```
Logger failed to save 1 LogEntryEvent__e records for [integration@mail.com] (Schema.User ID: 123)

Errors:
StatusCode: STRING_TOO_LONG, Message: Value too long for field, Field(s): [HttpRequestEndpoint__c]
```

**Comment by ILambuRI** [comment]
A temporary fix if you have the same problem:
![image](https://github.com/user-attachments/assets/68ed48d3-0a8f-4c72-b3fd-4c1ccf2095bf)

**Comment by jongpie** [comment]
Thanks @ILambuRI for reporting this! Just for my own reference, could you tell me how long the endpoint is in your particular example? You don't have to provide me with the actual value, but I'm curious exactly how long the endpoint is when you encounter this issue.

And since the existing endpoint field it too short, and it's a text(255) field, I can't increase the field length of the existing field - I think this will have to be fixed by introducing new long textarea fields (similar to what I did in [release `v4.13.15`](https://github.com/jongpie/NebulaLogger/releases/tag/v4.13.15) for the old `BrowserUrl__c` field):
- `HttpRequestEndpoint__c` --> replace with new long textarea `HttpRequestEndpointAddress__c`
- `HttpResponseEndpoint__c` --> replace with new long textarea `HttpResponseEndpointAddress__c`

For both `HttpRequestEndpoint__c` and `HttpResponseEndpoint__c`, they'll still be supported/populated with a truncated value for, but I'll consider them deprecated (and they'll eventually be removed, but I don't have a specific timeline for doing that).

**Comment by jongpie** [comment]
@ILambuRI I went ahead and released a fix for this in [release `v4.14.12`](https://github.com/jongpie/NebulaLogger/releases/tag/v4.14.12), with 2 main parts:
1. The value of `HttpRequestEndpoint__c` is now auto-truncated (same idea as what you show [in your comment](https://github.com/jongpie/NebulaLogger/issues/768#issuecomment-2363646018))
2. `HttpRequestEndpoint__c` is now considered deprecated (but still populated) - there's now a new field `LogEntry__c.HttpRequestEndpointAddress__c` that can store 2,000 characters

If it's easy for you to get the info, I'd still be curious to know the length of the endpoint you were using when you encountered this issue - my hope is that 2,000 characters should be plenty for storing endpoints/URLs, but I can still adjust the length of the field in future releases if needed.

**Comment by ILambuRI** [comment]
> @ILambuRI I went ahead and released a fix for this in [release `v4.14.12`](https://github.com/jongpie/NebulaLogger/releases/tag/v4.14.12), with 2 main parts:
> 
> 1. The value of `HttpRequestEndpoint__c` is now auto-truncated (same idea as what you show [in your comment](https://github.com/jongpie/NebulaLogger/issues/768#issuecomment-2363646018))
> 2. `HttpRequestEndpoint__c` is now considered deprecated (but still populated) - there's now a new field `LogEntry__c.HttpRequestEndpointAddress__c` that can store 2,000 characters
> 
> If it's easy for you to get the info, I'd still be curious to know the length of the endpoint you were using when you encountered this issue - my hope is that 2,000 characters should be plenty for storing endpoints/URLs, but I can still adjust the length of the field in future releases if needed.

Hi @jongpie , thx for the fast fix!
I'm using a pre-signed AWS S3 Bucket URL and it's around 1500-1600 characters.
![image](https://github.com/user-attachments/assets/70a91e31-5cf3-43bf-b67a-a367b8656bbd)

**Comment by jongpie** [comment]
@ILambuRI awesome, thanks for the info! It sounds like the 2,000 characters I used for `LogEntry__c.HttpRequestEndpointAddress__c` should be enough, so I'll leave the length as-is for now. If/when it needs to be increased further, I can do that in a separate release.

Thanks again for your help with this! Let me know if you run into any more issues.
